packages/cli/src/command-implementations/lint-command.ts

```diff
@@ -23,10 +23,37 @@ export class LintCommand extends ProjectCommand<LintCommandDefinition> {
   }
 
   private async _runLintForIntegration(definition: IntegrationDefinition): Promise<void> {
-    const parsedIntegrationDefinition = await prepareCreateIntegrationBody(definition, { dereference: true })
+    const strippedDefinition = this._stripAutoGeneratedContentFromIntegration(definition)
+    const parsedIntegrationDefinition = await prepareCreateIntegrationBody(strippedDefinition, { dereference: true })
     const linter = new IntegrationLinter(parsedIntegrationDefinition)
 
     await linter.lint()
     linter.logResults(this.logger)
   }
+
+  private _stripAutoGeneratedContentFromIntegration(definition: IntegrationDefinition) {
+    const { actionNames, eventNames } = this._getAutoGeneratedContentOfIntegration(definition)
+
+    return {
+      ...definition,
+      actions: Object.fromEntries(Object.entries(definition.actions ?? {}).filter(([key]) => !actionNames.has(key))),
+      events: Object.fromEntries(Object.entries(definition.events ?? {}).filter(([key]) => !eventNames.has(key))),
+    } as IntegrationDefinition
+  }
+
+  private _getAutoGeneratedContentOfIntegration(definition: IntegrationDefinition) {
+    const actionNames = new Set<string>()
+    const eventNames = new Set<string>()
+
+    for (const iface of Object.values(definition.interfaces)) {
+      for (const actionDefinition of Object.values(iface.actions)) {
+        actionNames.add(actionDefinition.name)
+      }
+      for (const eventDefinition of Object.values(iface.events)) {
+        eventNames.add(eventDefinition.name)
+      }
+    }
```

Comment:
this ugly loop could be simplified with flatMap, but I'm not a fan of iterating more than once over `definition.interfaces`:
```typescript
const actionNames = new Set(
  Object.values(definition.interfaces).flatMap(iface => 
    Object.values(iface.actions).map(action => action.name)
  )
)
const eventNames = new Set(
  Object.values(definition.interfaces).flatMap(iface => 
    Object.values(iface.events).map(event => event.name)
  )
)
```

Reply:
No I like this 👍  Good old procedural fashion is always easy to read and understand